src/Instrumentation/Laravel/src/LaravelInstrumentation.php

```diff
@@ -4,52 +4,81 @@
 
 namespace OpenTelemetry\Contrib\Instrumentation\Laravel;
 
+use Composer\InstalledVersions;
 use Illuminate\Contracts\Foundation\Application;
 use Illuminate\Foundation\Console\ServeCommand;
-use OpenTelemetry\API\Instrumentation\CachedInstrumentation;
+use OpenTelemetry\API\Globals;
+use OpenTelemetry\API\Instrumentation\InstrumentationInterface;
+use OpenTelemetry\API\Instrumentation\InstrumentationTrait;
 use OpenTelemetry\Contrib\Instrumentation\Laravel\Watchers\CacheWatcher;
 use OpenTelemetry\Contrib\Instrumentation\Laravel\Watchers\ClientRequestWatcher;
 use OpenTelemetry\Contrib\Instrumentation\Laravel\Watchers\ExceptionWatcher;
 use OpenTelemetry\Contrib\Instrumentation\Laravel\Watchers\LogWatcher;
 use OpenTelemetry\Contrib\Instrumentation\Laravel\Watchers\QueryWatcher;
 use OpenTelemetry\Contrib\Instrumentation\Laravel\Watchers\RequestWatcher;
-use OpenTelemetry\Contrib\Instrumentation\Laravel\Watchers\Watcher;
 use function OpenTelemetry\Instrumentation\hook;
 use Throwable;
 
-class LaravelInstrumentation
+class LaravelInstrumentation implements InstrumentationInterface
 {
-    public const NAME = 'laravel';
+    use InstrumentationTrait;
 
-    public static function registerWatchers(Application $app, Watcher $watcher)
+    public function getName(): string
     {
-        $watcher->register($app);
+        return 'io.opentelemetry.contrib.php.laravel';
     }
 
-    public static function register(): void
+    public function getVersion(): ?string
     {
-        $instrumentation = new CachedInstrumentation('io.opentelemetry.contrib.php.laravel');
+        return InstalledVersions::getVersion('open-telemetry/opentelemetry-auto-laravel');
+    }
+
+    public function getSchemaUrl(): ?string
+    {
+        return null;
+    }
+
+    public function init(): bool
+    {
+        $this->setPropagator(Globals::propagator());
+        $this->setTracerProvider(Globals::tracerProvider());
+        $this->setMeterProvider(Globals::meterProvider());
+
+        return true;
+    }
 
-        hook(
-            Application::class,
-            '__construct',
-            post: static function (Application $application, array $params, mixed $returnValue, ?Throwable $exception) use ($instrumentation) {
-                self::registerWatchers($application, new CacheWatcher());
-                self::registerWatchers($application, new ClientRequestWatcher($instrumentation));
-                self::registerWatchers($application, new ExceptionWatcher());
-                self::registerWatchers($application, new LogWatcher());
-                self::registerWatchers($application, new QueryWatcher($instrumentation));
-                self::registerWatchers($application, new RequestWatcher());
-            },
-        );
+    public function activate(): bool
+    {
+        if (!$this->init()) {
+            return false;
```

Comment:
I was unsure of the expected usage of this, based on [these comments](https://github.com/open-telemetry/opentelemetry-php/blob/e44f5a97d999e88de41aed7d54723354ea8505bf/src/API/Instrumentation/InstrumentationTrait.php#L100-L105).